### PR TITLE
Break up wait job tasks into processing and audit complete

### PR DESCRIPTION
**Pull request by ChomickiM**
IALERT-3067 

Added a new WaitJobCondition as well as calculating the average audit time

### CODE REVIEW

File: src/test/java/com/synopsys/integration/alert/performance/utility/AuditProcessingWaitJobTask.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package com.synopsys.integration.alert.performance.utility;
```
Comment (psantos1113) [inline src/test/java/com/synopsys/integration/alert/performance/utility/AuditProcessingWaitJobTask.java:1]: We will want to add a TODO to this class to refactor to be able to remove the duplicate code in the future.  This is functional but eventually we will want to refactor this.

**Review by psantos1113 — APPROVED** [review]

File: src/test/java/com/synopsys/integration/alert/performance/utility/AuditCompleteWaitJobTask.java [hunk 2]
```diff
@@ -154,4 +166,42 @@ private String createAuditRequestString(int pageNumber, int pageSize, Notificati
...
+            return;
+        }
+        intLogger.info(String.format("Performance: Average audit time: %s seconds.", averageAuditTime.getAsDouble()));
+    }
+
+    private Long calculateAuditTimeDifference(AuditJobStatusModel auditJobStatusModel) {
+        try {
+            OffsetDateTime timeCreated = DateUtils.parseDate(auditJobStatusModel.getTimeAuditCreated(), DateUtils.AUDIT_DATE_FORMAT);
+            OffsetDateTime timeLastSent = DateUtils.parseDate(auditJobStatusModel.getTimeLastSent(), DateUtils.AUDIT_DATE_FORMAT);
+            return ChronoUnit.SECONDS.between(timeCreated, timeLastSent);
```
Comment (psantos1113) [inline src/test/java/com/synopsys/integration/alert/performance/utility/AuditCompleteWaitJobTask.java:200]: Can you change the method to return a double and use the doubleValue() method of a Long object.

Comment (psantos1113) [reply src/test/java/com/synopsys/integration/alert/performance/utility/AuditCompleteWaitJobTask.java:200]: Thought of this after the fact

File: src/test/java/com/synopsys/integration/alert/performance/utility/AuditCompleteWaitJobTask.java [hunk 3]
```diff
@@ -96,13 +109,12 @@ private Set<String> getJobIdsFromAuditEntries() throws IntegrationException {
...
-            //            boolean anyPending = jobAuditModels.stream()
-            //                .anyMatch(Predicate.not(this::jobFinished));
-            //            if (anyPending) {
-            //                intLogger.info("Performance: Some audit entries are still processing. Continuing...");
-            //                return Set.of();
-            //            }
+            boolean anyPending = jobAuditModels.stream()
+                .anyMatch(Predicate.not(this::jobFinished));
+            if (anyPending) {
+                intLogger.info("Performance: Some audit entries are still processing. Continuing...");
```
Comment (bseifert14) [inline src/test/java/com/synopsys/integration/alert/performance/utility/AuditCompleteWaitJobTask.java:115]: Is it worth listing out the processing audit entries here? Something like:
```
if (anyPending) {
                intLogger.info("Performance: Some audit entries are still processing. Continuing...");
                forEach (pendingJobAuditModels) {
                            intLogger.info("Processing Job: %s", jobThatsProcessing);
                }
}
```

I'm not sure how granular we want to be - feel free to disregard!

**Pull request merged**